### PR TITLE
bonds: UI updates and import/export

### PR DESCRIPTION
**Pull request by chappjc**
This PR contains:

- some final UI updates for the fee-->bonds change
- includes bonds when doing account import/export, fix need to restart after import
- a few minor client/core fixes
- dep updates flagged by dependabot

### CODE REVIEW

**Review by martonp — COMMENTED** [review]
During registration, on the wallet funding screen, it allows you to only deposit the amount required for 1 bond, but then when it's time to post the bond, it will reject if there is not enough for the second bond. Either we should allow the user to not have to reserve enough for the second bond, or not go to the next page.

Another issue I found was when exporting the account. I exported, created a fresh dexc with the same seed, then imported. The imported server did not show up, but after I restarted the dexc, I could see it. Might be a pre-existing issue.

**Comment by chappjc** [comment]
> The imported server did not show up, but after I restarted the dexc, I could see it. Might be a pre-existing issue.

Yeah, this is the annoying `initializeDEXconnections` call.  I'll look at avoiding the restart but I think it might need to stay this way for now.

**Comment by chappjc** [comment]
> During registration, on the wallet funding screen, it allows you to only deposit the amount required for 1 bond, but then when it's time to post the bond, it will reject if there is not enough for the second bond. Either we should allow the user to not have to reserve enough for the second bond, or not go to the next page.

This was one of the things I was still considering.  You're absolutely right of course, with the current UI.  What I was pondering is a new user that might want to post a single bond *without* maintenance.  Seeing as we'll need to add a check box for that and probably handle other edges, I think we can put that idea on the back burner until 1.0 or maybe 0.6.1.  The normal user will have a maintained account. 

Will change the wallet wait form so it waits for the amount that it suggests, which accounts for the overlap with maintenance running.

File: client/webserver/locales/pl-pl.go [hunk 1]
```diff
@@ -98,108 +98,104 @@ var PlPL = map[string]string{
...
-	"confirmations":               "potwierdzenia",
-	"how_reg":                     "Jak chcesz uiścić opłatę rejestracyjną?",
-	"All markets at":              "Wszystkie rynki na",
-	"pick a different asset":      "wybierz inne aktywo",
-	"Create":                      "Utwórz",
-	"Register_loudly":             "Zarejestruj!",
-	"1 Sync the Blockchain":       "1: Zsynchronizuj blockchain",
-	"Progress":                    "Postęp",
-	"remaining":                   "pozostało",
-	"2 Fund the Registration Fee": "2: Wnieś opłatę rejestracyjną",
```
Comment (chappjc) [inline client/webserver/locales/pl-pl.go:177]: I couldn't see a way to tame gofmt here, not without leaving these incorrect translations.

File: client/webserver/locales/en-us.go [hunk 2]
```diff
@@ -25,29 +25,29 @@ var EnUS = map[string]string{
 	"remove":                         "remove",
 	"add a file":                     "add a file",
 	"Submit":                         "Submit",
-	"Skip Registration":              "Skip Registration (view-only mode)",
-	"Confirm Registration":           "Confirm Registration",
-	"app_pw_reg":                     "Enter your app password to confirm DEX registration.",
-	"reg_confirm_submit":             `When you submit this form, funds will be spent from your wallet to pay the registration fee.`,
+	"Skip Registration":              "No account (view-only mode)",
```
[the remark this answers is not in the document]
Comment (chappjc) [reply client/webserver/locales/en-us.go:28]: Would be nice if we had a "stale" flag like with the notifications because this one isn't bad still.

**Comment by chappjc** [comment]
image

image

image

image

image
(I missed the conf countdown).

image

File: client/webserver/locales/en-us.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chappjc) [reply client/webserver/locales/en-us.go:28]: Removed some and flagged some others with comments if they were still accurate.

**Comment by chappjc** [comment]
The need to restart after import should also be resolved @martonp but I've not tested that.  If the required changes somehow turn out to be much bigger we'll have to punt on this.

**Comment by martonp** [comment]
The import basically works now, but if there was an active bond then it is not able to do a refund 

```
2023-03-16 15:47:05.836 [WRN] CORE: Bond output not found, possibly already spent or never mined! Marking refunded. Backup refund transaction: 
2023-03-16 15:47:05.836 [ERR] CORE: Failed to mark bond as refunded: bond bucket does not exist: 376aae28be249c5c0494b21433d5df1144085b50cf957a3a2cba850e72cba8ac
```
I think this will require a more extensive change.

**Comment by martonp** [comment]
It's not a huge deal, but I think the text `You may deposit as much as you like` might make people think you can deposit just the 10 DCR. Maybe below the `Bond Amount: 10 DCR` there could also be `Minimum Deposit: 20.01` in the same size text.

**Comment by chappjc** [comment]
> The import basically works now, but if there was an active bond then it is not able to do a refund
> 
> ```
> 2023-03-16 15:47:05.836 [WRN] CORE: Bond output not found, possibly already spent or never mined! Marking refunded. Backup refund transaction: 
> 2023-03-16 15:47:05.836 [ERR] CORE: Failed to mark bond as refunded: bond bucket does not exist: 376aae28be249c5c0494b21433d5df1144085b50cf957a3a2cba850e72cba8ac
> ```
> 
> I think this will require a more extensive change.

Was there really no refund tx in the log or did you remove it for brevity?  I don't see why it wouldn't be exported and imported given

```go
RefundTx   []byte `json:"refundTx"`
```

but it really needs to be.

**Comment by chappjc** [comment]
Oh wait, I think the server may have reported an unknown bond there.  I guess I'll toy with this some more.

**Comment by chappjc** [comment]
Ah I need to update the `async exportAccount () {` in dexsettings.ts.  The JSON that the browser downloads lacks bonds.

**Comment by chappjc** [comment]
Yeesh, finally got bond import working.  General cases:

- Restore from same seed, discover account done, then import account used to import previous bonds.  The bond key index will derive the correct private key.  We need to update any placeholder live and unspent bonds when importing, otherwise append new unspent bonds that are likely expired and need refund.
- New seed used, registered fresh, then import account used to import previous bonds.  The bonds to not apply to the account, so we must block this.  Further, the bond key index will NOT derive the correct private key.  The user must manually broadcast the refund tx from the JSON file.
- New seed used, import the account with its bonds.  The bonds do apply to the account, but as with the previous case the bond key index will NOT derive the correct private key, so automatic refund must use the backup refund tx (in `rotateBonds`).  However, this pays to a the old wallet.  This is a messy case and should only be considered as a recovery aid.

In the last two points above, this is an unfortunate consequence of how we switched from storing the bond private key in the DB to storing the key index since that is only helpful if the app seed is unchanged.  Since the backup refund tx pays to the old wallet, this is suboptimal, but at least they can restore the wallet if they had the original seed.

**Review by martonp — APPROVED** [review]
LGTM, all the import cases work well. Could optionally export the bond private keys so they can be used when imported with a dexc that was initialized with another seed. It's probably a niche use case though so probably not worth the time.

**Review by buck54321 — CHANGES_REQUESTED** [review]
I sent the exact recommended amount, but the registration sequence did not progress. I had so send more.
image

File: client/core/account.go [hunk 3]
```diff
@@ -138,17 +140,70 @@ func (c *Core) AccountImport(pw []byte, acct *Account, bonds []*db.Bond) error {
...
+			}
+		}
+		if newLiveBonds == 0 {
+			return nil
+		}
+		c.log.Infof("Imported %d new unspent bonds", newLiveBonds)
+		if dc, connected, _ := c.dex(host); connected {
+			c.disconnectDEX(dc)
+			// TODO: less heavy handed approach to append or update
+			// dc.acct.{bonds,pendingBonds,expiredBonds}, using server config...
```
[the remark this answers is not in the document]
Comment (chappjc) [reply client/core/account.go:198]: add or update.  Probably other error prone stuff.

File: client/webserver/site/src/js/forms.ts [hunk 4]
```diff
@@ -726,8 +726,12 @@ export class ConfirmRegistrationForm {
     this.pwCache = pwCache
 
     Doc.bind(this.page.goBack, 'click', () => goBack())
+    // bondStrengthField is presently hidden since there is no scaling of user
+    // limits yet, and there needs to be considerable explanation of why
```
[the remark this answers is not in the document]
Comment (chappjc) [reply client/webserver/site/src/js/forms.ts:730]: "there needs to be considerable explanation" --> "there needs for considerable explanation"?

[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/forms.ts:730]: Oh. Hmm. I don't know how I was reading this.

**Review by buck54321 — APPROVED** [review]

**Pull request merged**